agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/classicsdk/BytecodeUtilImpl.java

```diff
@@ -67,10 +69,22 @@ public void setConnectionString(String connectionString) {
               + " \"connectionStringConfiguredAtRuntime\" to true");
       return;
     }
-    if (TelemetryClient.getActive().getConnectionString() != null) {
-      logger.warn("Connection string is already set");
-      return;
+    if (connectionStringProgrammaticallySet.getAndSet(true)) {
+      String applicationInsightsConnectionStringEnv =
```

Comment:
Since you make `APPLICATIONINSIGHTS_CONNECTION_STRING_ENV` public, you can import the constant directly. I don't think it's necessary to create a local variable.

Reply:
should we just check `!TelemetryClient.getActive().getConnectionString().equals(System.getenv(APPLICATIONINSIGHTS_CONNECTION_STRING_ENV)`?

Reply:
static import of constant, you simply use 'APPLICATIONINSIGHTS_CONNECTION_STRING_ENV', a string constant.

Reply:
I don't mind

Reply:
It's just preferences

Reply:
> should we just check `!TelemetryClient.getActive().getConnectionString().equals(System.getenv(APPLICATIONINSIGHTS_CONNECTION_STRING_ENV)`?

this is similar to my original thought, but

`!TelemetryClient.getActive().getConnectionString().equals(connectionString)`

Reply:
The current verification of the connection string value is applied [only once](https://github.com/microsoft/ApplicationInsights-Java/blob/f15c27664daacb6a02d7ce47812693d01371c5b6/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/classicsdk/BytecodeUtilImpl.java#L72) to avoid many logs in cases where the application often changes the connection string. The connection string from the environment variable could be equal to the programmatic connection string value when the connection is programmatically set the first time but not  the following times.  For these reasons, the current version verifies that a connection string from an environment variable exists.

Reply:
```
if (setConnectionStringFirstTimeCalled && TelemetryClient.getActive().getConnectionString() != null) {
  // info log that programmatic call is overriding value from env var / json
}
```